docs/spec/concepts.rst

```diff
@@ -3,6 +3,266 @@
+consistent with ``A``, if and only if there exists some fully static type ``C``
+which is a materialization of both ``A`` and ``B``.
+
+``Any`` is consistent with every type, and every type is consistent with
+``Any``. (This follows from the definitions of materialization and consistency
+but is worth stating explicitly.)
+
+The consistency relation is not transitive. ``tuple[int, int]`` is consistent
+with ``tuple[Any, int]``, and ``tuple[Any, int]`` is consistent with
+``tuple[str, int]``, but ``tuple[int, int]`` is not consistent with
+``tuple[str, int]``.
+
+The consistency relation is symmetric. If ``A`` is consistent with ``B``, ``B``
+is also consistent with ``A``. It is also reflexive: ``A`` is always consistent
+with ``A``.
+
+The assignable-to (or consistent subtyping) relation
+----------------------------------------------------
+
+Given the materialization relation and the subtyping relation, we can define the
+**consistent subtype** relation over all types. A type ``A`` is a consistent
+subtype of a type ``B`` if there exists a materialization ``A'`` of ``A`` and a
+materialization ``B'`` of ``B``, where ``A'`` and ``B'`` are both static types,
+and ``A'`` is a subtype of ``B'``.
+
+Consistent subtyping defines "assignability" for Python.  An expression can be
+assigned to a variable (including passed as a parameter or returned from a
+function) if it is a consistent subtype of the variable's type annotation
+(respectively, parameter's type annotation or return type annotation).
+
+We can say that a type ``A`` is "assignable to" a type ``B`` if ``A`` is a
+consistent subtype of ``B``.
+
+In the remainder of this specification, we will usually prefer the term
+**assignable to** over "consistent subtype of". The two are synonymous, but
+"assignable to" is shorter, and may communicate a clearer intuition to many
+readers.
+
+For example, ``Any`` is assignable to ``int``, because ``int`` is a
+materialization of ``Any``, and ``int`` is a subtype of ``int``. The same
+materialization also gives that ``int`` is assignable to ``Any``.
+
+The assignable-to relation is not generally symmetric, though. If ``B`` is a
+subtype of ``A``, then ``tuple[Any, B]`` is assignable to ``tuple[int, A]``,
+because ``tuple[Any, B]`` can materialize to ``tuple[int, B]``, which is a
+subtype of ``tuple[int, A]``. But ``tuple[int, A]`` is not assignable to
+``tuple[Any, B]``.
+
+Attributes and methods
+----------------------
+
+In Python, we can do more with objects at runtime than just assign them to
+names, pass them to functions, or return them from functions. We can also
+get/set attributes and call methods.
+
+In the Python object model, the operations that can be performed on a value all
+de-sugar to method calls. For example, ``a + b`` is syntactic sugar for either
+``a.__add__(b)`` or ``b.__radd__(a)``.
+
+For a static type checker, accessing ``a.foo`` is a type error unless all
```

Comment:
This statement isn't strictly true if some of the objects support `__getattr__` (or set / delete depending on the context). I'm not sure if it's worth muddying the description with this caveat though. Just wanted to point it out in case you want to do something with it.

Reply:
Thanks! Added a parenthetical about `__getattr__` to avoid misinterpretation.

I think it would be good for the spec to have a deeper discussion of attribute getters and setters and their types (and how this works with gradual and set-theoretic types), but this will be a quite nuanced topic, so I'd rather tackle it in a later PR.

Reply:
Yes, I agree this deserves its own section.

FWIW, here are some [notes](https://github.com/erictraut/mypy_issues/blob/main/tc_topics.md) that I've been maintaining for myself about the typing spec and conformance tests. In these notes (which are admittedly rough and use a shorthand that may not make sense to others), I list some ideas for additional chapters and topics that I think are missing from the current spec.